/**
 * SPL-218038: Patch bootstrap-2.3.1.min.js to remediate
 * XSS Security Vulnerability CVE-2018-14040
 * Patched web/search_mrsparkle/exposed/js/contrib/bootstrap-2.3.1.min.js
 * https://nvd.nist.gov/vuln/detail/CVE-2018-14040
 * https://github.com/twbs/bootstrap/pull/23687/files
 */

--- docs/assets/js/bootstrap.js	2022-04-27 21:28:31.000000000 -0700
+++ docs/assets/js/bootstrap.js.ORIG	2022-04-27 21:15:46.000000000 -0700
 
 !function ($) {
@@ -100,7 +87,7 @@
       selector = selector && selector.replace(/.*(?=#[^\s]*$)/, '') //strip for ie7
     }
 
-    $parent = $(selector)
+    $parent = $(document).find(selector)
 
     e && e.preventDefault()
 
@@ -452,7 +439,8 @@
 
   $(document).on('click.carousel.data-api', '[data-slide], [data-slide-to]', function (e) {
     var $this = $(this), href
-      , $target = $($this.attr('data-target') || (href = $this.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '')) //strip for ie7
+      , target = $($this.attr('data-target') || (href = $this.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '')) //strip for ie7
+      , $target = $(document).find(target)
       , options = $.extend({}, $target.data(), $this.data())
       , slideIndex
 
@@ -626,9 +614,10 @@
       , target = $this.attr('data-target')
         || e.preventDefault()
         || (href = $this.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') //strip for ie7
-      , option = $(target).data('collapse') ? 'toggle' : $this.data()
-    $this[$(target).hasClass('in') ? 'addClass' : 'removeClass']('collapsed')
-    $(target).collapse(option)
+      , $target = $(document).find(target)
+      , option = $target.data('collapse') ? 'toggle' : $this.data()
+    $this[$target.hasClass('in') ? 'addClass' : 'removeClass']('collapsed')
+    $target.collapse(option)
   })
 
 }(window.jQuery);/* ============================================================
@@ -751,7 +740,7 @@
       selector = selector && /#/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') //strip for ie7
     }
 
-    $parent = selector && $(selector)
+    $parent = selector && $(document).find(selector)
 
     if (!$parent || !$parent.length) $parent = $this.parent()
 
@@ -1030,7 +1019,8 @@
   $(document).on('click.modal.data-api', '[data-toggle="modal"]', function (e) {
     var $this = $(this)
       , href = $this.attr('href')
-      , $target = $($this.attr('data-target') || (href && href.replace(/.*(?=#[^\s]+$)/, ''))) //strip for ie7
+      , target = $($this.attr('data-target') || (href && href.replace(/.*(?=#[^\s]+$)/, ''))) //strip for ie7
+      , $target = $(document).find(target)
       , option = $target.data('modal') ? 'toggle' : $.extend({ remote:!/#/.test(href) && href }, $target.data(), $this.data())
 
     e.preventDefault()
@@ -1624,7 +1614,7 @@
           + '[data-target="' + target + '"],'
           + this.selector + '[href="' + target + '"]'
 
-        active = $(selector)
+        active = $(document).find(selector)
           .parent('li')
           .addClass('active')
 
@@ -1740,7 +1730,7 @@
 
       if (e.isDefaultPrevented()) return
 
-      $target = $(selector)
+      $target = $(document).find(selector)
 
       this.activate($this.parent('li'), $ul)
       this.activate($target, $target.parent(), function () {
 to remediate
 * XSS Security Vulnerability CVE-2018-14040
 * Patched web/search_mrsparkle/exposed/js/contrib/bootstrap-2.3.1.min.js
 * https://nvd.nist.gov/vuln/detail/CVE-2018-14040
 * https://github.com/twbs/bootstrap/pull/23687/files
 */

--- docs/assets/js/bootstrap.js	2022-04-27 21:28:31.000000000 -0700
+++ docs/assets/js/bootstrap.js.ORIG	2022-04-27 21:15:46.000000000 -0700
 
 !function ($) {
@@ -100,7 +87,7 @@
       selector = selector && selector.replace(/.*(?=#[^\s]*$)/, '') //strip for ie7
     }
 
-    $parent = $(selector)
+    $parent = $(document).find(selector)
 
     e && e.preventDefault()
 
@@ -452,7 +439,8 @@
 
   $(document).on('click.carousel.data-api', '[data-slide], [data-slide-to]', function (e) {
     var $this = $(this), href
-      , $target = $($this.attr('data-target') || (href = $this.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '')) //strip for ie7
+      , target = $($this.attr('data-target') || (href = $this.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '')) //strip for ie7
+      , $target = $(document).find(target)
       , options = $.extend({}, $target.data(), $this.data())
       , slideIndex
 
@@ -626,9 +614,10 @@
       , target = $this.attr('data-target')
         || e.preventDefault()
         || (href = $this.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') //strip for ie7
-      , option = $(target).data('collapse') ? 'toggle' : $this.data()
-    $this[$(target).hasClass('in') ? 'addClass' : 'removeClass']('collapsed')
-    $(target).collapse(option)
+      , $target = $(document).find(target)
+      , option = $target.data('collapse') ? 'toggle' : $this.data()
+    $this[$target.hasClass('in') ? 'addClass' : 'removeClass']('collapsed')
+    $target.collapse(option)
   })
 
 }(window.jQuery);/* ============================================================
@@ -751,7 +740,7 @@
       selector = selector && /#/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') //strip for ie7
     }
 
-    $parent = selector && $(selector)
+    $parent = selector && $(document).find(selector)
 
     if (!$parent || !$parent.length) $parent = $this.parent()
 
@@ -1030,7 +1019,8 @@
   $(document).on('click.modal.data-api', '[data-toggle="modal"]', function (e) {
     var $this = $(this)
       , href = $this.attr('href')
-      , $target = $($this.attr('data-target') || (href && href.replace(/.*(?=#[^\s]+$)/, ''))) //strip for ie7
+      , target = $($this.attr('data-target') || (href && href.replace(/.*(?=#[^\s]+$)/, ''))) //strip for ie7
+      , $target = $(document).find(target)
       , option = $target.data('modal') ? 'toggle' : $.extend({ remote:!/#/.test(href) && href }, $target.data(), $this.data())
 
     e.preventDefault()
@@ -1624,7 +1614,7 @@
           + '[data-target="' + target + '"],'
           + this.selector + '[href="' + target + '"]'
 
-        active = $(selector)
+        active = $(document).find(selector)
           .parent('li')
           .addClass('active')
 
@@ -1740,7 +1730,7 @@
 
       if (e.isDefaultPrevented()) return
 
-      $target = $(selector)
+      $target = $(document).find(selector)
 
       this.activate($this.parent('li'), $ul)
       this.activate($target, $target.parent(), function () {
